### PR TITLE
feat(bundle-manager): blacklisted bundle directory names

### PR DESCRIPTION
**Pull request by haganbmj**
Allows irrelevant folders (like `.git`) and invalid bundles to exist in the bundles directory without crashing.
It's likely that the bundle-parser should be logging these errors and returning an undefined bundle instead to avoid using this blind try/catch. That'd mean passing in and constructing a new logger instance each time with the current implementation, though.

### CODE REVIEW

**Comment by haganbmj** [comment]
Not sure there's a way to get this to rerun the security checks.

**Comment by Lange** [comment]
NodeCG itself is failing security checks on master so it's okay that the PR is failing.

Can you elaborate on the goals of this PR, and what specific problems it solves? I don't quite 100% understand it yet. NodeCG refusing to start when invalid bundles are present is currently intentional, though I'm open to feedback on that design decision.

**Comment by haganbmj** [comment]
Yeah, maybe I'm just trying to solve a problem I created for myself when I decided that my bundles directory should be one big git repo of submodules.

Without this commit nodecg exits because `.git` is an invalid bundle.

**Comment by Lange** [comment]
I could see other people wanting to do that. I'll go ahead and merge this.

**Comment by haganbmj** [comment]
There might be a better way though, I don't doubt that at all. Maybe just a couple blacklisted directory names and the ability to add a few exclusions in your `cfg/node.json`

**Comment by Lange** [comment]
I could go for that too. Blacklisting `.git` by default seems reasonable.

**Comment by haganbmj** [comment]
Whoops, was able to get some new issues related to chokidar because the watchers don't adhere to the ignored directories.

Unsure what the symlink limitations are, wasn't even sure nodecg resolved symlinks, but it's probably alright to just move this up into the bundle discovery loop. Whether that's before or after the semver compatibility check or not.
https://github.com/nodecg/nodecg/blob/c39d45cd3066f48e09369c5dc2ef132fe24d945f/lib/bundle-manager.js#L190-L194

```
2018-05-02 10:38 NodeCG[43514] (FSEvents.framework) process_dir_events: creation: watch_path() failed for '~/git/nodecg/bundles/.git/dashboard'
2018-05-02 10:38 NodeCG[43514] (FSEvents.framework) process_dir_events: creation: watch_path() failed for '~/git/nodecg/bundles/.git/.git'
2018-05-02 10:38 NodeCG[43514] (FSEvents.framework) process_dir_events: creation: watch_path() failed for '~/git/nodecg/bundles/.git/.git'
2018-05-02 10:38 NodeCG[43514] (FSEvents.framework) process_dir_events: creation: watch_path() failed for '~/git/nodecg/bundles/.git/dashboard'
2018-05-02 10:39 NodeCG[43514] (FSEvents.framework) process_dir_events: creation: watch_path() failed for '~/git/nodecg/bundles/.git/dashboard'
2018-05-02 10:39 NodeCG[43514] (FSEvents.framework) process_dir_events: creation: watch_path() failed for '~/git/nodecg/bundles/.git/.git'
```

File: lib/bundle-manager.js [hunk 1]
```diff
@@ -140,6 +146,11 @@ module.exports.init = function (bundlesPath, cfgPath, nodecgVersion, nodecgConfi
 			return;
 		}
 
+		// Prevent attempting to load unwanted directories. Those specified above and all dot-prefixed.
+		if (blacklistedBundleDirectories.indexOf(bundleFolderName) > -1 || bundleFolderName.startsWith('.')) {
```
Comment (Lange) [inline lib/bundle-manager.js]: `blacklistedBundleDirectories.indexOf(bundleFolderName) > -1` can be simplified to `blacklistedBundleDirectories.includes(bundleFolderName)`

**Comment by haganbmj** [comment]
Change:

Add the concept of blacklisted bundle directories, directories that can safely exist in your `bundles` directory without nodecg attempting to load them.

My use case for this was having my `bundles` directory managed as a git repository with submodules. Due to this the `.git` directory was being parsed as a bundle and nodecg would exception out.

The currently ignored directories are the common things like `node_modules` and `bower_components`, but also any dot-prefixed directories (like `.vscode`, `.idea`).


As for a status, the ignoring works fine. There's some changes that need to be made to the chokidar logic though because it bypasses the concept of enabled/disabled bundles entirely and watches every folder in the bundles directory.

**Pull request merged**